src/store/ducks/opensrp/hierarchies/index.tsx

```diff
@@ -438,37 +431,17 @@ export const getMetaForPlanInTree = () => {
   });
 };
 
-/** combine tree with metaData
- * @param state - the store
- * @param props - the filterProps
- */
-export const getTreeWithMeta = () => {
-  return createSelector(getTreeById(), getMetaForPlanInTree(), (origTree, metaForTree) => {
-    // walk the tree and update meta field.
-    if (!origTree) {
-      return;
-    }
-    const treeClone = cloneDeep(origTree);
-    treeClone.walk(node => {
-      node.model.meta = metaForTree[node.model.id] ? metaForTree[node.model.id] : {};
-      return true;
-    });
-
-    return treeClone;
-  });
-};
-
 /** retrieve a node given the id
  * @param state - the store
  * @param props -  the filterProps
  */
 export const getNodeById = () =>
-  createSelector(getTreeWithMeta(), getNodeId, (tree, nodeId) => {
+  createSelector(getTreeById(), getNodeId, getMetaForPlanInTree(), (tree, nodeId, metaData) => {
     if (!tree || !nodeId) {
       return;
     }
     const nodeOfInterest = tree.first(node => node.model.id === nodeId);
```

Comment:
What are the implications of using `.find` here?

Reply:
.first is a method of the tree ADT we are using. I do not think it supports find.